code/edgeidentity/src/main/java/com/adobe/marketing/mobile/edge/identity/IdentityState.java

```diff
@@ -28,24 +28,20 @@
  */
 class IdentityState {
 
+	private static final String LOG_SOURCE = "IdentityState";
+
+	private final IdentityStorageManager identityStorageManager;
 	private IdentityProperties identityProperties;
 	private boolean hasBooted;
 
 	/**
 	 * Loads the persisted identities (if any) into {@link #identityProperties}
 	 */
-	IdentityState() {
-		this(IdentityStorageService.loadPropertiesFromPersistence());
-	}
+	IdentityState(final IdentityStorageManager identityStorageManager) {
```

Comment:
thanks for the update. It looks like this constructor is no longer needed then, because you can use the default constructor with mocks now

Reply:
A large part of logic in this class depends on operations on `IdentityStorageManager` . 
```
identityStorageManager.saveToPersistence() 
identityStorageManager.loadPropertiesFromPersistence()
identityStorageManager.loadEcidFromDirectIdentityPersistence()

``` 
It is possible to mock `DataStoring` to provide necessary values for the properties loaded from persistence. However, IMO it would mean that the `IdentityStateTests` will have internal knowledge on how IdentityStorageService uses `DataStoring` (i.e one level deeper than what is used in `IdentityState`) as opposed to treating the `IdentityStorageManager` as a black box mock. It follows that the test mock logic for `IdentityStateTests` will need to change if the internal implementation of `IdentityStorageManager` changes (keeping the api same) making these unit tests not so isolated anymore. 
So my inclination would be to retain this cascading test constructor for test isolation. Let me know what you think!

Also, thinking about it - it would just be sufficient to change the default constructor as :
```
	IdentityState() {
		this(new IdentityStorageManager(ServiceProvider.getInstance().getDataStoreService()));
	}

        @VisibleForTesting
	IdentityState(final IdentityStorageManager identityStorageManager) {
                 .......
        }

```

Reply:
My thinking was that you can spare a mock and use the mocked datastore service for both the classes since there is no much secret on the main operations the 3 methods do. But if you want to have more control over the mocks for unit testing we can make IdentityStorageManager mockable as you described.

Reply:
Thanks! Yea, I prefer having limited logic in test setup so I have made the storage manager mockable as of now. I am definitely open to revisiting this later in test PR if we see that that the mock data store have better value.